airflow-core/src/airflow/exceptions.py

```diff
@@ -21,121 +21,66 @@
 
 from __future__ import annotations
 
-from collections.abc import Collection, Sequence
-from datetime import datetime, timedelta
 from http import HTTPStatus
-from typing import TYPE_CHECKING, Any, NamedTuple
+from typing import TYPE_CHECKING, NamedTuple
 
 if TYPE_CHECKING:
     from airflow.models import DagRun
-    from airflow.sdk.definitions.asset import AssetNameRef, AssetUniqueKey, AssetUriRef
-    from airflow.utils.state import DagRunState
 
 # Re exporting AirflowConfigException from shared configuration
 from airflow._shared.configuration.exceptions import AirflowConfigException as AirflowConfigException
 
+try:
+    from airflow.sdk.exceptions import (
+        AirflowException,
+        AirflowNotFoundException,
+        AirflowRescheduleException,
+        TaskNotFound,
+    )
+except ModuleNotFoundError:
+    # When _AIRFLOW__AS_LIBRARY is set, airflow.sdk may not be installed.
+    # In that case, we define fallback exception classes that mirror the SDK ones.
```

Comment:
```suggestion
    # Compatibility for airflow versions < 3.2
    # We define fallback exception classes that mirror the SDK ones
```

Reply:
This one is about _AIRFLOW__AS_LIBRARY cause that's only when we could have this issue.  Not compat for versions <3.2.